[quant][graphmode] Run RemoveRedundantDequantize in the end

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #40038 [quant][graphmode] Support another use pattern of mean
* #39919 [quant] Support general op modules with inplace options
* #39925 [quant][graphmode] Support quantizing `repeat`
* #39924 [quant][graphmode] Support squeeze/unsqueeze
* **#39923 [quant][graphmode] Run RemoveRedundantDequantize in the end**
* #39915 [quant][graphmode] Pass debug option into insert_quant_dequant pass
* #39911 [quant][graphmode] Use quantized::batch_norm in graph mode
* #39910 [quant] add quantized::batch_norm
* #39895 [quant][graphmode] Support prim::TupleUnpack and prim::TupleConstruct

Summary:
`propagateQuantizationOps` generate new `dequantize` ops, so this pass should run after that.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22014712](https://our.internmc.facebook.com/intern/diff/D22014712)

jerryzh168 committed:
[quant][graphmode] Run RemoveRedundantDequantize in the end

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Run RemoveRedundantDequantize in the end"


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22014712](https://our.internmc.facebook.com/intern/diff/D22014712)

[ghstack-poisoned]

vkuzo reviewed (approved):
lg, would be awesome if we also had some context on the "why" (for new team members to learn)

jerryzh168 commented:
> lg, would be awesome if we also had some context on the "why" (for new team members to learn)

sure, updated summary

jerryzh168 committed:
Update on "[quant][graphmode] Run RemoveRedundantDequantize in the end"


Summary:
`propagateQuantizationOps` generate new `dequantize` ops, so this pass should run after that.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22014712](https://our.internmc.facebook.com/intern/diff/D22014712)

[ghstack-poisoned]
